cmd/etcd.go

```diff
@@ -107,6 +107,10 @@ func SnapshotSaveEtcdHosts(
 	flags cluster.ExternalFlags, snapshotName string) error {
 
 	log.Infof(ctx, "Starting saving snapshot on etcd hosts")
+
+	stateFilePath := cluster.GetStateFilePath(flags.ClusterFilePath, flags.ConfigDir)
+	rkeFullState, _ := cluster.ReadStateFile(ctx, stateFilePath)
```

Comment:
why are we always eating this error instead of returning? it's in several places and just seems weird. ok if it's on purpose but everywhere i saw it we could return it...

Reply:
Because the choice was made to seamlessly upgrade from RKE v0.1.x to v0.2.x which introduced the statefile, so if it's not there, it doesn't mean an error as it can be a cluster to be upgraded to contain/support a statefile. Now we know this probably could have been better but we also have to think of Rancher where we can't interact with cluster provisioning (e.g. ask the user a question before proceeding)